tools/oeedger8r/CMakeLists.txt

```diff
@@ -1,6 +1,76 @@
 # Copyright (c) Open Enclave SDK contributors.
 # Licensed under the MIT License.
 
+find_program(OCAMLOPT ocamlopt)
+
+if (OCAMLOPT)
+  set(BINARY ${CMAKE_CURRENT_BINARY_DIR}/oeedger8r${CMAKE_EXECUTABLE_SUFFIX})
+  add_custom_command(
+    OUTPUT ${BINARY}
+    # Copy sources to build folder.
+    COMMAND cmake -E copy ${CMAKE_CURRENT_SOURCE_DIR}/intel/*
+            ${CMAKE_CURRENT_BINARY_DIR}
+    COMMAND cmake -E copy ${CMAKE_CURRENT_SOURCE_DIR}/src/*
+            ${CMAKE_CURRENT_BINARY_DIR}
+    # Generate lexer and parser.
+    COMMAND ocamllex Lexer.mll
+    COMMAND ocamlyacc Parser.mly
+    # Generate Intel library.
+    # Use -w -3 to suppress
+    # warning 3: deprecated: String.lowercase
+    COMMAND
+      ocamlopt -c -bin-annot -for-pack Intel -w -3 Ast.ml Util.ml SimpleStack.ml
+      Plugin.ml Preprocessor.ml Parser.mli Parser.ml Lexer.ml CodeGen.ml
+    COMMAND ocamlopt -pack -o Intel.cmx Ast.cmx Util.cmx SimpleStack.cmx
```

Comment:
Can you split the source files into list variables? i.e. "INTEL_SOURCES" and "OE_SOURCES". Then you can use those in compilation and DEPENDS.

Reply:
I second this. However, currently this command depends on the files and compiles their copies in `CMAKE_CURRENT_BINARY_DIR`. The prefix differences need to be handled if using list.

Reply:
I tried to create a list but it did not work out due to the reason @mingweishih points out.
- The sources have to be copied to bin dir; otherwise ocamlopt will pollute the source folder by generating artifacts there.
- Some sources (Parser.ml, Parser.mli, Lexer.ml) are generated on the fly from Parser.ml, Lexer.ml
- The DEPENDS declaration must have relative paths to the original source locations - they must have the intel/src prefix.

All this makes using a list of files a bit difficult.